dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,306 @@
+                final SecretKeyEntry secretKeyEntry = (SecretKeyEntry) ks.getEntry(variableKey, keyStorePP);
+                return ((PBEKeySpec) factory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class)).getPassword();
+            } else {
+                return CACHE_404;
+            }
+
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+
+    @Override
+    public List<String> listKeys() {
+        final KeyStore keyStore = getSecretsStore();
+        return Sneaky.sneak(() -> Collections.list(keyStore.aliases()));
+    }
+
+    @Override
+    public synchronized boolean deleteAll(){
+        final File secretStoreFile = new File(secretsKeyStorePath);
+        secretStoreFile.delete();
+        flushCache();
+        return true;
+
+    }
+
+
+    /**
+     * This method saves a secret value into the keystore. The value to be stored is encrypted using the
+     * company key and is used as a "password" for a keystore entry - using it as a passowrd allows us
+     * to store an arbitrary string in a keystore
+     * 
+     * @param variableKey
+     * @param variableValue
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     * @throws KeyStoreException
+     * @throws Exception
+     */
+    @Override
+    public boolean saveValue(final String variableKey, final char[] variableValue) {
+        try {
+            final SecretKeyFactory factory = SecretKeyFactory.getInstance(
+                    SECRETS_STORE_SECRET_KEY_FACTORY_TYPE);
+            final SecretKey generatedSecret = factory.generateSecret(new PBEKeySpec(encrypt(variableValue).toCharArray()));
+            final KeyStore keyStore = getSecretsStore();
+            final PasswordProtection keyStorePP = new PasswordProtection(loadStorePassword());
+            keyStore.setEntry(variableKey, new KeyStore.SecretKeyEntry(generatedSecret), keyStorePP);
+            saveSecretsStore(keyStore);
+            flushCache(variableKey);
+        } catch (Exception e) {
+            Logger.warn(this.getClass(), "Unable to save secret from " + SECRETS_STORE_FILE + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+        return true;
+    }
+
+    /**
+     * deletes a value from the store
+     */
+    @Override
+    public void deleteValue(final String secretKey) {
+        try {
+            final KeyStore keyStore = getSecretsStore();
+            keyStore.deleteEntry(secretKey);
+            saveSecretsStore(keyStore);
+            flushCache(secretKey);
+        } catch (KeyStoreException e) {
+            Logger.warn(this.getClass(), "Unable to delete secret from  " + SECRETS_STORE_FILE + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private Key key() {
+        return Sneaky.sneak(() -> APILocator.getCompanyAPI().getDefaultCompany()).getKeyObj();
+    }
+
+    @VisibleForTesting
+    protected String encrypt(final char[] val) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)